lib/publiccloud/utils.pm

```diff
@@ -260,6 +266,20 @@ sub get_ssh_private_key_path {
     return (is_azure() || is_openstack() || get_var('PUBLIC_CLOUD_LTP')) ? "~/.ssh/id_rsa" : '~/.ssh/id_ed25519';
 }
 
+sub permit_root_passwordless {
+    my ($instance) = @_;
+
+    # Permit root passwordless login over SSH
+    $instance->ssh_assert_script_run('sudo cat /etc/ssh/sshd_config');
+    $instance->ssh_assert_script_run('sudo sed -i "s/PermitRootLogin no/PermitRootLogin prohibit-password/g" /etc/ssh/sshd_config');
+    $instance->ssh_assert_script_run('sudo sed -i "/^AllowTcpForwarding/c\AllowTcpForwarding yes" /etc/ssh/sshd_config') if (is_hardened());
+    $instance->ssh_assert_script_run('sudo systemctl reload sshd');
+
+    # Copy SSH settings for remote root
+    $instance->ssh_assert_script_run('sudo install -o root -g root -m 0700 -dD /root/.ssh');
+    $instance->ssh_assert_script_run(sprintf("sudo install -o root -g root -m 0644 /home/%s/.ssh/authorized_keys /root/.ssh/", $instance->{username}));
+}
+
```

Comment:
I'd keep those generic settings as they are because we run them everywhere and just do 'special' cloud-init testing for cloud-init. Is that ok? Or what do you suggest?

Reply:
Sure, keep them. I am just saying that sshd setup should be done by cloud-init as it is how the users would do it. If you use a drop-in config file, instaed of mock `write_files` and `run_cmd` we can use them to configure sshd.